bot/internal/review/review.go

```diff
@@ -197,13 +197,20 @@ func (r *Assignments) getReleaseReviewers() []string {
 	return r.c.ReleaseReviewers
 }
 
-func (r *Assignments) getDocsReviewers(author string) []string {
-	setA, setB := getReviewerSets(author, "Core", r.c.DocsReviewers, r.c.DocsReviewersOmit)
-	reviewers := append(setA, setB...)
+func (r *Assignments) getDocsReviewers(e *env.Environment, files []github.PullRequestFile) []string {
+	// See if any code reviewers are designated preferred reviewers for one of
+	// the changed docs files. If so, add them as docs reviewers.
+	a, b := getReviewerSets(e.Author, "Core", r.c.CodeReviewers, r.c.CodeReviewersOmit)
+	prefCodeReviewers := r.getAllPreferredReviewers(append(a, b...), files)
+
+	// Get the docs reviewer pool, which does not depend on the files
+	// changed by a pull request.
+	docsA, docsB := getReviewerSets(e.Author, "Core", r.c.DocsReviewers, r.c.DocsReviewersOmit)
+	reviewers := append(prefCodeReviewers, append(docsA, docsB...)...)
 
 	// If no docs reviewers were assigned, assign admin reviews.
 	if len(reviewers) == 0 {
-		return r.getAdminReviewers(author)
+		return r.getAdminReviewers(e.Author)
```

Comment:
I know this isn't related to your changes, but while we're here I think it would be nice to log cases like this where we fall back to admin reviewers.